python/desc/twinkles/phosim_cpu_pred.py

```diff
@@ -0,0 +1,91 @@
+"""
+Estimates the CPU time required for a phosim simulation of a 30-s visit.  The
+inputs are filter, moonalt, and moonphase, or obsHistID (an Opsim ID from
+a specified (hard coded) Opsim sqlite database.
+
+The random forest is generated (and saved as a pickle file) by 
+run1_cpu_generate_rf.py, using only the filter, moonalt, and moonphase
+features.  This script needs to be run only once.
+"""
+
+import pickle
+import math
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import pylab
+from desc.twinkles.sqlite_tools import SqliteDataFrameFactory
+
+#print('Loading Random forest pickle')
+#RFbest = pickle.load(open("RF_pickle.p", "rb"))
+#print('Pickle loaded')
+
+class CpuPred(object):
+    """
+    Returns predicted fell-class CPU time in seconds for user-supplied
+    filter (0-5 for ugrizy), moon altitude (deg), and moon phase (0-100) 
+    -or- ObsHistID from the kraken_1042 Opsim run.  The code is hardwired to
+    look for the sqlite database for kraken_1042 in a specific location on
+    SLAC Linux.
+
+    RF_pickle.p is written by run1_cpu_generate_rf.py
+    """
+    def __init__(self, rf_pickle_file='RF_pickle.p', opsim_db_file='/nfs/farm/g/lsst/u1/DESC/Twinkles/kraken_1042_sqlite.db'):
+        self.RFbest = pickle.load(open(rf_pickle_file, 'rb'))
+        self.factory = SqliteDataFrameFactory(opsim_db_file)
+    def __call__(self, filter_arg, moonalt_arg, moonphase_arg):
+        return 10.**self.RFbest.predict(np.array([[filter_arg, moonalt_arg, 
+            moonphase_arg]]))
+    def opsimdb(self, obsid):
+        rec = self.factory.create('obsHistID filter moonAlt moonPhase'.split(), 
+            'Summary', 
+            condition="where obsHistID='%i' order by obsHistID asc"%obsid)
+
+        if rec.size <> 0:
+            # Translate the filter string into an index 0-5
+            filtmap = 'u g r i z y'.split()
+            filt = filtmap.index(rec["filter"].values[0])
```

Comment:
Ah, good :-)
I have not yet noticed what pylint makes of my "literate python" style,
which aims to construct python statements that read as grammatically
correct English.

On Thu, May 12, 2016 at 5:18 PM, James Chiang <notifications@github.com>
wrote:

> In python/desc/twinkles/phosim_cpu_pred.py
> <https://github.com/DarkEnergyScienceCollaboration/Twinkles/pull/240#discussion_r63117907>
> :
>
> > +    """
> > +    def __init__(self, rf_pickle_file='RF_pickle.p', opsim_db_file='/nfs/farm/g/lsst/u1/DESC/Twinkles/kraken_1042_sqlite.db'):
> > +        self.RFbest = pickle.load(open(rf_pickle_file, 'rb'))
> > +        self.factory = SqliteDataFrameFactory(opsim_db_file)
> > +    def __call__(self, filter_arg, moonalt_arg, moonphase_arg):
> > +        return 10.**self.RFbest.predict(np.array([[filter_arg, moonalt_arg,
> > +            moonphase_arg]]))
> > +    def opsimdb(self, obsid):
> > +        rec = self.factory.create('obsHistID filter moonAlt moonPhase'.split(),
> > +            'Summary',
> > +            condition="where obsHistID='%i' order by obsHistID asc"%obsid)
> > +
> > +        if rec.size <> 0:
> > +            # Translate the filter string into an index 0-5
> > +            filtmap = 'u g r i z y'.split()
> > +            filt = filtmap.index(rec["filter"].values[0])
>
> It's fine Phil. pylint will sometimes complain about too many local
> variables (which is itself annoying, but the acceptable number is
> configurable), so I've tried to be more concise recently, relying on common
> idioms (common in my estimation, at least) to provide clarity. Being very
> explicit in variable usage is fine.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/DarkEnergyScienceCollaboration/Twinkles/pull/240/files/e5d1e7836483140ba3eb5e77dcc297b5e9ab3b47#r63117907>
>